module/VuFind/src/VuFind/Db/AdapterFactory.php

```diff
@@ -233,26 +240,68 @@ public function getAdapterFromConnectionString(
         $username = $overrideUser ?? $username;
         $password = $overridePass ?? $password;
 
-        $driverName = $this->getDriverName($type);
-        $driverOptions = $this->getDriverOptions($driverName);
-
-        // Set up default options:
-        $options = [
-            'driver' => $driverName,
+        return $this->getAdapterFromArray([
+            'driver' => $type,
             'hostname' => $host,
             'username' => $username,
             'password' => $password,
             'database' => $dbName,
             'use_ssl' => $this->config->Database->use_ssl ?? false,
-            'driver_options' => $driverOptions,
+            'port' => $port ?? null,
+        ]);
+    }
+
+    /**
+     * Obtain a Laminas\DB connection using the config.
+     *
+     * @param string $overrideUser Username override (leave null to use username from config)
+     * @param string $overridePass Password override (leave null to use password from config)
+     *
+     * @return Adapter
+     */
+    public function getAdapterFromConfig($overrideUser = null, $overridePass = null)
+    {
+        if (!isset($this->config->Database)) {
+            throw new \Exception('"database" settings missing');
+        }
+        $config = $this->config->Database;
+        return $this->getAdapterFromArray([
+            'driver' => $config->database_driver ?? null,
+            'hostname' => $config->database_host ?? null,
+            'username' => $overrideUser ?? $config->database_username ?? null,
+            'password' => $overridePass ?? $this->getSecretFromConfig($config, 'database_password'),
+            'database' => $config->database_name,
```

Comment:
This should have a check that it exists particularly because it's a new one.

Reply:
With the last modifications, it defaults to the regular key in the config if it doesn't find the secret
Is that good enough?

Reply:
Yes, thanks!